BridgeCareApp/VuejsApp/src/services/scenario.service.ts

```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
         return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
-            axiosInstance.get('api/GetScenarios')
-                .then((responseLegacy: AxiosResponse<Scenario[]>) => {
-                    if (hasValue(responseLegacy)) {
-                        var resultant: Scenario[] = [];
-                        responseLegacy.data.forEach(simulation => {
-                            if (!any(propEq('simulationId', simulation.simulationId), scenarios)) {
-                                simulation.shared = false;
-                                simulation.status = 'N/A';
-                                resultant.push(simulation);
-                            }
-                        });
+            scenarios.forEach(simulation => {
+                nodejsAxiosInstance.delete(`api/DeleteMongoScenario/${(simulation as any)._id}`)
+                    .then((res: AxiosResponse) => {
+                        if (hasValue(res)) {
+                            return resolve(res);
+                        }
+                    })
+                    .catch((error: any) => {
+                        return resolve(error.response);
+                    });
+            });
+        });
+    }
 
-                        if (resultant.length != 0) {
-                            nodejsAxiosInstance.post('api/AddMultipleScenarios', resultant)
-                                .then((res: AxiosResponse<Scenario[]>) => {
+    /**
+     * Updates mongodb to match the scenarios from the legacy database
+     */
+    static getLegacyScenarios(): AxiosPromise {
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+            axiosInstance.get('api/GetScenarios')
+                .then((responseLegacy: AxiosResponse<Scenario[]>)=>{
+                    if (hasValue(responseLegacy)){
+                    this.getMongoScenarios()
+                        .then((responseMongo: AxiosResponse<Scenario[]>)=>{
+                            if (hasValue(responseMongo)) {
+                                var scenariosToAdd = this.getMissingScenarios(responseLegacy.data, responseMongo.data);
+                                var scenariosToRemove = this.getMissingScenarios(responseMongo.data, responseLegacy.data);
+                                if (scenariosToAdd.length != 0) {
+                                    nodejsAxiosInstance.post('api/AddMultipleScenarios', scenariosToAdd)
+                                        .then((res: AxiosResponse<Scenario[]>) => {
+                                            if (hasValue(res)){
+                                                return resolve(res);
+                                            }
+                                        })
+                                        .catch((error: any) => resolve(error.response));
+                                }
+                                this.removeMongoScenarios(scenariosToRemove).then((res: AxiosResponse) => {
```

Comment:
What happens if there are no scenarios to remove? Might we need a logical check to see if we need to run this step?

Reply:
The removeMongoScenarios function does nothing when given an empty array.

Should I add an explicit check around the call for the sake of readability?

Reply:
If it isn't causing an issues and is already not running if the array is empty then we can leave it as is.